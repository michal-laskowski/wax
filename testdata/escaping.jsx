
export function escaping(){
    return (<>
        <in-inner>
            <sample-01>{'"<>&"\''}</sample-01>
            <sample-02>{"'<>&\"'"}</sample-02>
            <sample-03><script>alert('sample-03')</script></sample-03>
            <sample-04>{"<script>alert('in-inner-sample-04')</script>"}</sample-04>
            <sample-05>{`<script>alert("in-inner-sample-05")</script>`}</sample-05>
            <sample-06 title={"'\"<>&"} style={{ textAlign: "'\"<>&" }}>{"'\"<>&"}</sample-06>
            <sample-07>{"<script type='' src=\"\"></script>"}</sample-07>
        </in-inner>
        <in-attr>
            <sample-01 v={'"<>&"\''} />
            <sample-02 v={"'<>&\"'"} />
            <sample-03 xstyle={"\"&<>'"} />
            <sample-04 style={{ backgroundColor: "\"&<>'" }} />
            <sample-05 class={"\"&<>'"} />
            <sample-06 class={'test:1" xss="false'} />
            <sample-07 onclick={`<script>alert("xdd")</script>`}>in-attr-sample-07</sample-07>
            <sample-08 v="'bar'" />
            <sample-09 v='"bar\"' />
            <sample-10 v="bar\`" />
            <sample-11 v={"'bar'"} />
            <sample-12 v={'"bar\`\"\''} />
            <sample-13 v={"bar\`\"\'"} />
        </in-attr>
    </>)
}