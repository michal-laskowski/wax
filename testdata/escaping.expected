<in-inner>
    <sample-01>&#34;&lt;&gt;&amp;&#34;&#39;</sample-01>
    <sample-02>&#39;&lt;&gt;&amp;&#34;&#39;</sample-02>
    <sample-03>
        <script>alert('sample-03')</script>
    </sample-03>
    <sample-04>
        &lt;script&gt;alert(&#39;in-inner-sample-04&#39;)&lt;/script&gt;
    </sample-04>
    <sample-05>
        &lt;script&gt;alert(&#34;in-inner-sample-05&#34;)&lt;/script&gt;
    </sample-05>
    <sample-06
        title="&#39;&#34;&lt;&gt;&amp;"
        style="text-align: &#39;\22&lt;&gt;&amp;">
        &#39;&#34;&lt;&gt;&amp;
    </sample-06>
    <sample-07>
        &lt;script type=&#39;&#39; src=&#34;&#34;&gt;&lt;/script&gt;
    </sample-07>
</in-inner>
<in-attr>
    <sample-01 v="&#34;&lt;&gt;&amp;&#34;&#39;"></sample-01>
    <sample-02 v="&#39;&lt;&gt;&amp;&#34;&#39;"></sample-02>
    <sample-03 xstyle="&#34;&amp;&lt;&gt;&#39;"></sample-03>
    <sample-04 style="background-color: \22&amp;&lt;&gt;&#39;"></sample-04>
    <sample-05 class="&#34;&amp;&lt;&gt;&#39;"></sample-05>
    <sample-06 class="test:1&#34; xss=&#34;false"></sample-06>
    <sample-07 onclick="&lt;script&gt;alert(&#34;xdd&#34;)&lt;/script&gt;">
        in-attr-sample-07
    </sample-07>
    <sample-08 v="&#39;bar&#39;"></sample-08>
    <sample-09 v="&#34;bar\&#34;"></sample-09>
    <sample-10 v="bar\`"></sample-10>
</in-attr>